### PR TITLE
Added docs in storeProtectLoadUtils.js, Docs added for loadProtector …

### PR DESCRIPTION
**Pull request by rabahalishah**
Contribution in docs branch: I have added few lines of documentation for loadProtector reducer function for having the a clear understanding for contributors about what does this reducer function is doing in middlerware.

### CODE REVIEW

**Comment by rabahalishah** [comment]
Its my first ever contribution kindly have a look into it.

**Comment by stevepiercy** [comment]
@rabahalishah please follow https://6.docs.plone.org/contributing/first-time.html

**Comment by rabahalishah** [comment]
@stevepiercy I have followed all rules. Like it seems simple. I have forked this repo and created a new branch called docs and then made some changes (added docs) and then commit and push it the docs branch and then made a pull request. With descriptive message of my contribution.
Can you please tell me where I am missing the rules which I should not violate?

**Comment by stevepiercy** [comment]
https://6.docs.plone.org/contributing/first-time.html#plone-contributors-team

**Comment by rabahalishah** [comment]
@stevepiercy I apologize to bother you Steve. But I have read the provided in the given link. I don't think so I am missing something. Can you please elaborate the main issue?

**Comment by rabahalishah** [comment]
@stevepiercy 
image

See my changes. please let me know. If you see any issue.

**Comment by stevepiercy** [comment]
> New contributors should wait for confirmation that they have been added to this team before creating a pull request to a Plone project.

Furthermore, you have not followed this:

https://6.docs.plone.org/contributing/first-time.html#requirements

> ## Requirements
> All first-time contributors to Plone must follow the contributing requirements described in [Contributing to Plone](https://6.docs.plone.org/contributing/index.html).

What about those items is not clear?

**Review by rabahalishah — COMMENTED** [review]
All looks fine.

**Comment by rabahalishah** [comment]
@stevepiercy Hello Steve, Now I am a member of plone foundation. I have signed the agreement yesterday. But still my contribution is blocked. can you please look into this matter?

**Comment by stevepiercy** [comment]
You are now showing as a member of the [plone/contributors](https://github.com/orgs/plone/teams/contributors) team.

Please see the details for each of the failing checks for more information.

The Towncrier and changelog checks fail due to this: https://6.docs.plone.org/contributing/index.html#change-log-entry

For Prettier, your changes do not pass linting checks. See https://6.docs.plone.org/volto/contributing/index.html for details.

File: packages/generator-volto/.eslintrc.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 module.exports = {
   // extends: ['eslint:recommended', 'prettier'],
-  plugins: ['prettier'],
-  rules: {
-    'prettier/prettier': 'error',
-  },
+  // plugins: ['prettier'],
+  // rules: {
+  //   'prettier/prettier': 'error',
+  // },
```
Comment (stevepiercy) [inline packages/generator-volto/.eslintrc.js:6]: Do not comment out tools that are required for CI. Please revert.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: packages/generator-volto/.eslintrc.js (hunk 1 above)
Comment (rabahalishah) [reply packages/generator-volto/.eslintrc.js:6]: Dont worry, This was just for testing purposes.

**Comment by rabahalishah** [comment]
@stevepiercy Hello Steve, I am very close. My changes have successfully passed 18 test only four are failing. I don't know why they are still failing I have followed all steps mentioned in Plone's contribution guideline. Can you please have a look into my plone [repo](https://github.com/rabahalishah/volto)? Please help me in fixing these failing.

**Comment by stevepiercy** [comment]
@rabahalishah each failing workflow has a link for Details, which usually has sufficient information to take action, such as this one: https://github.com/plone/volto/actions/runs/5937157687/job/16099010883?pr=5104#step:5:66

You also need to make sure that this branch is up to date with `master`, because all workflows are based on that branch.

File: src/middleware/storeProtectLoadUtils.js [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-const GET_CONTENT_FAIL = 'GET_CONTENT_FAIL';
-const RESET_CONTENT = 'RESET_CONTENT';
+const LOCATION_CHANGE = "@@router/LOCATION_CHANGE";
+const PROTECT_START = "@@loadProtector/START";
+const PROTECT_END = "@@loadProtector/END";
+const PROTECT_SKIPPED = "@@loadProtector/SKIPPED";
+const GET_CONTENT_PENDING = "GET_CONTENT_PENDING";
+const GET_CONTENT_SUCCESS = "GET_CONTENT_SUCCESS";
+const GET_CONTENT_FAIL = "GET_CONTENT_FAIL";
+const RESET_CONTENT = "RESET_CONTENT";
```
Comment (stevepiercy) [inline src/middleware/storeProtectLoadUtils.js:10]: Revert all the above formatting changes and configure your editor not to convert quotes.

File: package.json [hunk 3]
```diff
@@ -419,6 +418,7 @@
     "jest": "26.6.3",
     "jest-environment-jsdom": "^26",
     "jsonwebtoken": "9.0.0",
+    "prettier": "3.0.2",
```
Comment (stevepiercy) [inline package.json:421]: Why did you change this? This may be the cause to one failed workflow. I would suggest that you revert this change, if you do not know what it does.

File: packages/generator-volto/package.json [hunk 4]
```diff
@@ -124,7 +124,7 @@
     "husky": "^3.0.9",
     "jest": "^24.8.0",
     "lint-staged": "^9.4.3",
-    "prettier": "2.0.5",
+    "prettier": "3.0.2",
```
Comment (stevepiercy) [inline packages/generator-volto/package.json:127]: Same.

File: packages/generator-volto/news/5104.documentation [hunk 5]
```diff
@@ -0,0 +1 @@
+Added documention for loadProtector reducer function in storeProtectLoadUtils.js Middlerware. @rabahalishah
```
Comment (stevepiercy) [inline packages/generator-volto/news/5104.documentation:1]: Spelling and MyST formatting.

```suggestion
Added documentation for `loadProtector` reducer function in `storeProtectLoadUtils.js` middleware. @rabahalishah
```

File: packages/generator-volto/yarn.lock [hunk 6]
```diff
@@ -8460,12 +8460,12 @@ __metadata:
...
-  version: 2.0.5
-  resolution: "prettier@npm:2.0.5"
+"prettier@npm:3.0.2":
+  version: 3.0.2
+  resolution: "prettier@npm:3.0.2"
   bin:
-    prettier: bin-prettier.js
-  checksum: 7f89d2f5d1a1a15a7bf200556b8c5395913d0119fe52a3fba51ab116695d584652d37fdb7d86e0919bfe36b22649afcf222eaca00a1065eb486b7b4cf3062eff
+    prettier: bin/prettier.cjs
+  checksum: 118b59ddb6c80abe2315ab6d0f4dd1b253be5cfdb20622fa5b65bb1573dcd362e6dd3dcf2711dd3ebfe64aecf7bdc75de8a69dc2422dcd35bdde7610586b677a
```
Comment (stevepiercy) [inline packages/generator-volto/yarn.lock:8468]: Again, I don't know why you changed this verion.

File: src/middleware/storeProtectLoadUtils.js [hunk 7]
```diff
@@ -21,83 +21,95 @@ const RESET_CONTENT = 'RESET_CONTENT';
...
+        () =>
+          dispatch({
+            type: PROTECT_END,
+          }),
+        0
+      );
+    }
+    return next(action);
+  };
+
```
Comment (stevepiercy) [inline src/middleware/storeProtectLoadUtils.js:100]: Revert all the above reformatting changes. They may be the cause of failing workflows.

File: yarn.lock [hunk 8]
```diff
@@ -18895,12 +18895,12 @@ __metadata:
...
+"prettier@npm:3.0.2":
+  version: 3.0.2
+  resolution: "prettier@npm:3.0.2"
   bin:
-    prettier: bin-prettier.js
-  checksum: 7f89d2f5d1a1a15a7bf200556b8c5395913d0119fe52a3fba51ab116695d584652d37fdb7d86e0919bfe36b22649afcf222eaca00a1065eb486b7b4cf3062eff
+    prettier: bin/prettier.cjs
+  checksum: 118b59ddb6c80abe2315ab6d0f4dd1b253be5cfdb20622fa5b65bb1573dcd362e6dd3dcf2711dd3ebfe64aecf7bdc75de8a69dc2422dcd35bdde7610586b677a
   languageName: node
   linkType: hard
```
Comment (stevepiercy) [inline yarn.lock:18905]: Revert all the above changes, as they may be the cause of failed workflows.

File: packages/generator-volto/yarn.lock (hunk 6 above)
Comment (rabahalishah) [reply packages/generator-volto/yarn.lock:8468]: Sorry to say @stevepiercy, I have not changed anything showing above. I just simply added docs and did nothing. I don't know how this has happened.

Comment (stevepiercy) [reply packages/generator-volto/yarn.lock:8468]: hmm, so this was not you, but someone masquerading as you?

https://github.com/plone/volto/blame/17e96908d99696b3e631184c733c0fa4edc8dac1/packages/generator-volto/yarn.lock#L8467-L8468

It's just a mistake, no big deal. If you don't know why something changes, then do not `git add` the changed files and push them. Also pay careful attention to your editor if it suggests that you add files to a commit that you did not intentionally change. Finally, do not just `git add` from the command line because that adds every file that you changed, added, and deleted, whether you realize it or not.
